substrate/dex/pallet/src/lib.rs

```diff
@@ -311,6 +334,38 @@ pub mod pallet {
 
   #[pallet::hooks]
   impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+    fn on_finalize(n: BlockNumberFor<T>) {
+      for coin in Pools::<T>::iter_keys() {
+        // insert the new price to our oracle window
+        let last = Self::quote_price_exact_tokens_for_tokens(coin, Coin::native(), 1, false)
```

Comment:
1 isn't a valid quantity here. It'll lose all precision. Any coin worth less than SRI will always return 0.

I'd recommend using as many decimals as the external coin has (we used to have this metadata back with pallet-assets, but I believe it was removed/moved to some primitives?).

Reply:
I'd also document the security considerations behind using the end of block price re: preventing mid-block spikes a malicious block proposer can exploit without risk of being arbitraged against.